### PR TITLE
CUMULUS-2635: Fix data-migration2 granules migration with PDR changes

### PR DESCRIPTION
**Pull request by botanical**
**Summary:** Summary of changes

Addresses [CUMULUS-2635: Investigate granule migration failures that reference PDRs](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2635)

## Changes

- Update `data-migration2` to migrate PDRs before migrating granules
## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: lambdas/data-migration2/src/pdrs.ts [hunk 1]
```diff
@@ -62,41 +54,11 @@ export const migratePdrRecord = async (
...
-    pan_message: dynamoPDR.PANmessage,
-    stats: dynamoPDR.stats,
-    address: dynamoPDR.address,
-    original_url: dynamoPDR.originalUrl,
-    timestamp: dynamoPDR.timestamp ? new Date(dynamoPDR.timestamp) : undefined,
-    duration: dynamoPDR.duration,
-    created_at: new Date(dynamoPDR.createdAt),
-    updated_at: dynamoPDR.updatedAt ? new Date(dynamoPDR.updatedAt) : undefined,
-  };
+  const updatedRecord = await translateApiPdrToPostgresPdr(
```
Comment (markdboyd) [inline lambdas/data-migration2/src/pdrs.ts:58]: Very well done 👏

File: packages/db/src/translate/pdrs.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ * Generate a Postgres PDR record from a DynamoDB record.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} dynamoRecord
+ *   Record from DynamoDB
+ * @param {Knex | Knex.Transaction} knexOrTransaction
+ *   Knex client for reading from RDS database
+ * @param {Object} collectionPgModel - Instance of the collection database model
+ * @param {Object} providerPgModel   - Instance of the provider database model
+ * @param {Object} executionPgModel  - Instance of the execution database model
+ * @returns {Object} A PDR PG record
```
Comment (markdboyd) [inline packages/db/src/translate/pdrs.ts:20]:
```suggestion
 * @returns {PostgresPdr} A PDR PG record
```

File: lambdas/data-migration2/tests/test-index.js [hunk 3]
```diff
@@ -153,21 +156,21 @@ test.serial('handler migrates executions, granules, files, and PDRs by default',
...
+    { name: testPdr.pdrName }
   );
   t.is(
-    granuleRecords.length,
+    pdrRecords.length,
     1
   );
 
-  const pdrRecords = await t.context.pdrPgModel.search(
+  const granuleRecords = await t.context.granulePgModel.search(
```
Comment (markdboyd) [inline lambdas/data-migration2/tests/test-index.js:168]: This is all great, but I wonder if we should verify that the `pdr_cumulus_id` on `granuleRecords[0]` matches the `pdrRecords[0].cumulus_id`? And use the `GranulesExecutionsPgModel.search` to determine if this a record joining the granule cumulus ID and execution cumulus ID?

Just to be sure that our migration code is doing the right thing and creating references from the granule to those other records

**Review by markdboyd — CHANGES_REQUESTED** [review]
Great work! I only have a couple pretty minor suggestions

File: lambdas/data-migration2/tests/test-index.js (hunk 3 above)
Comment (botanical) [reply lambdas/data-migration2/tests/test-index.js:168]: That's a good point! I will update that to check

Comment (botanical) [reply lambdas/data-migration2/tests/test-index.js:168]: [`94d2489` (#2402)](https://github.com/nasa/cumulus/pull/2402/commits/94d2489dff0ece950d4cbe1518a7052c2566e5f8)

File: packages/db/src/translate/pdrs.ts (hunk 2 above)
Comment (botanical) [reply packages/db/src/translate/pdrs.ts]: [`94d2489` (#2402)](https://github.com/nasa/cumulus/pull/2402/commits/94d2489dff0ece950d4cbe1518a7052c2566e5f8)

File: lambdas/data-migration2/tests/test-index.js (hunk 3 above)
Comment (markdboyd) [reply lambdas/data-migration2/tests/test-index.js:168]: It seems like we didn't address this part:

> And use the GranulesExecutionsPgModel.search to determine if this a record joining the granule cumulus ID and execution cumulus ID?

**Review by markdboyd — APPROVED** [review]

**Pull request merged**